src/ucp/wireup/select.c

```diff
@@ -763,19 +768,21 @@ static void ucp_wireup_fill_peer_err_criteria(ucp_wireup_criteria_t *criteria,
 static void ucp_wireup_fill_aux_criteria(ucp_wireup_criteria_t *criteria,
                                          unsigned ep_init_flags)
 {
-    criteria->title              = "auxiliary";
-    criteria->local_md_flags     = 0;
-    criteria->remote_md_flags    = 0;
-    criteria->local_iface_flags  = UCT_IFACE_FLAG_CONNECT_TO_IFACE |
-                                   UCT_IFACE_FLAG_AM_BCOPY |
-                                   UCT_IFACE_FLAG_PENDING;
-    criteria->remote_iface_flags = UCT_IFACE_FLAG_CONNECT_TO_IFACE |
-                                   UCT_IFACE_FLAG_AM_BCOPY |
-                                   UCT_IFACE_FLAG_CB_ASYNC;
-    criteria->local_event_flags  = 0;
-    criteria->remote_event_flags = 0;
-    criteria->calc_score         = ucp_wireup_aux_score_func;
-    criteria->tl_rsc_flags       = UCP_TL_RSC_FLAG_AUX; /* Can use aux transports */
+    criteria->title                   = "auxiliary";
+    criteria->local_md_flags          = 0;
+    criteria->remote_md_flags         = 0;
+    criteria->local_iface_flags       = UCT_IFACE_FLAG_AM_BCOPY |
+                                        UCT_IFACE_FLAG_PENDING;
+    criteria->remote_iface_flags      = UCT_IFACE_FLAG_AM_BCOPY;
+    if (!ucp_ep_init_flags_has_cm(ep_init_flags)) {
+        criteria->local_iface_flags  |= UCT_IFACE_FLAG_CONNECT_TO_IFACE;
```

Comment:
pls add a comment that if no `CONNECT_TO_*` flag set  - we can select any type (if it works really this way)